plugins/gradle/java/src/execution/test/runner/AllInPackageGradleConfigurationProducer.java

```diff
@@ -73,7 +84,12 @@ protected boolean doIsConfigurationFromContext(ExternalSystemRunConfiguration co
 
     final String scriptParameters = configuration.getSettings().getScriptParameters() + ' ';
     final String filter = createTestFilterFrom(configurationData.psiPackage, /*hasSuffix=*/true);
-    return scriptParameters.contains(filter);
+    if (!scriptParameters.contains(filter)) return false;
+    VirtualFile locationFile = (VirtualFile)configuration.getUserData(LOCATION_NAME);
+    // If locationFile is null, then return true. This is for all the existing configurations created without location information
+    // from previous versions, as they would have already passed all the previous comparisons.
+    if (locationFile == null) return true;
```

Comment:
I am not sure why you would like to match run configurations based on `locationFile`, but please be aware that `get/setUserData`  is not persistent and may have desired effect after IDEA restart.

Reply:
Please ignore that. I forgot to clean a very old commit. Th PR is just about a small change now. PTAL and let me know if looks good !